Source/OxyPlot/Series/ExtrapolationLineSeries.cs

```diff
@@ -0,0 +1,322 @@
+        {
+            double previous = minX;
+
+            if (this.orderedIntervals != null && this.orderedIntervals.Any())
+            {
+                IEnumerable<double> flatLimits
+                    = this.Flatten(this.orderedIntervals).Where(l => l >= minX && l <= maxX);
+
+                foreach (double limiter in flatLimits)
+                {
+                    yield return new OxyRect(
+                        this.Transform(previous, minY),
+                        this.Transform(limiter, maxY))
+                        .Clip(clippingRect);
+
+                    previous = limiter;
+                }
+            }
+
+            yield return new OxyRect(
+                this.Transform(previous, minY),
+                this.Transform(maxX, maxY))
+                .Clip(clippingRect);
+        }
+
+        /// <summary>
+        /// Returns a flat sequence of doubles containing alternating minima
+        /// and maxima of the original data range intervals.
+        /// </summary>
+        private IEnumerable<double> Flatten(IEnumerable<DataRange> intervals)
+        {
+            foreach (var interval in intervals)
+            {
+                yield return interval.Minimum;
+                yield return interval.Maximum;
+            }
+        }
+
+        /// <summary>
+        /// Renders the part of the line which is given by the provided list of screen points.
+        /// </summary>
+        private void RenderLinePart(IRenderContext rc, IList<ScreenPoint> pointsToRender, bool isInterval)
+        {
+            OxyColor color = isInterval ? this.ExtrapolationColor : this.Color;
+
+            double[] dashes = isInterval ?
+                this.ActualExtrapolationDashArray :
+                this.ActualDashArray;
+
+            rc.DrawReducedLine(
+                pointsToRender,
+                this.MinimumSegmentLength * this.MinimumSegmentLength,
+                this.GetSelectableColor(color),
+                this.StrokeThickness,
+                this.EdgeRenderingMode,
+                dashes,
+                this.LineJoin);
+        }
+
+        /// <summary>
+        /// Sorts the intervals by their minimum and merges those intervals which overlap, i.e.
+        /// replaces them by their union.
+        /// </summary>
+        private List<DataRange> MergeOverlaps(IEnumerable<DataRange> intervals)
+        {
+            List<DataRange> orderedList = new List<DataRange>();
+
+            if (intervals != null)
+            {
+                IOrderedEnumerable<DataRange> ordered = intervals.OrderBy(i => i.Minimum);
+
+                foreach (DataRange current in ordered)
+                {
+                    DataRange previous = orderedList.LastOrDefault();
+
+                    if (current.IntersectsWith(previous))
+                    {
+                        orderedList[orderedList.Count - 1] = new DataRange(previous.Minimum, current.Maximum);
```

Comment:
max values should be `Math.Max(previous.Maximum, current.Maximum)`, otherwise a wider interval may 'consume' a smaller interval.

Reply:
Fixed.